scripts/install-nix-from-closure.sh

```diff
@@ -69,13 +60,17 @@ while [ $# -gt 0 ]; do
         --no-modify-profile)
             NIX_INSTALLER_NO_MODIFY_PROFILE=1;;
         --darwin-use-unencrypted-nix-store-volume)
-            CREATE_DARWIN_VOLUME=1;;
+            (
+                echo "Warning: the flag --darwin-use-unencrypted-nix-store-volume"
+                echo "         is no longer needed and will be removed in the future."
+                echo ""
```

Comment:
It may not be necessary but I'm not sure if that means we should remove support for it.

Reply:
Can you clarify?

Reply:
I'm just thinking someone who doesn't care about the volume being encrypted might prefer not having a launch daemon to deal with. But it's probably better to have only a single installation configuration, which means removing support for this flag is correct.

Reply:
Gotcha. 

By this point, it's fairly compatible with the existing install practice (in early drafts I had removed the fstab step to save us an install step and the user an uninstall step, but emily pointed out in #nix-darwin IRC that diskutil mount will pull mount options from it). Removing the scare-flag is an indication that I feel like solving the restore race-condition gets us to the point where a random Mac user can run the installer and it'll do something I could personally defend with a straight face if it ended up tending on HN.

I'm open to providing a no-daemon path here (though I feel like it's best to make an angry mob demand it, for the ~simplicity reason you mention). One option I've mentioned is just y/n prompting users to accept the daemon (but installing it by default if headless), which gives us a place (outside of the docs) to make it clear what problems it might cause, and that they'll have to manually fix their install if they regret it later. 

After subtracting the daemon, there's little technical difference between this PR and what we currently have, but I see having it as the default and having it as an opt-in fallback as a big qualitative difference! I think it's akin to Nix giving you *options* to do something radical like move "/nix", but putting it in a well-supported location by default.

Reply:
I think we should definitely investigate the reliability of simply granting access to `APFSUserAgent` (and `CSUserAgent`) for the keychain item and skipping the daemon. If that works, and if we believe that's reliable going forward, then that would greatly simplify things. I just don't know how to prove that this is future-compatible without getting feedback from an Apple engineer on this topic.

Reply:
Oops I just saw your top-level comment on this.